src/main.rs

```diff
@@ -0,0 +1,272 @@
+use hyper::{
+    Request,
+    {body::Bytes, client::conn::http1},
+};
+use hyper_util::rt::TokioIo;
+use jwt_simple::prelude::*;
+use log::{debug, error, info};
+use std::io::Write;
+use tokio::{net::UnixStream, process::Command};
+
+const TOKEN_EXPIRE_HOURES: u64 = 2;
+
+#[actix_web::main]
+async fn main() {
+    let mut builder;
+    log_panics::init();
+
+    if cfg!(debug_assertions) {
+        builder = Builder::from_env(Env::default().default_filter_or("debug"));
+    } else {
+        builder = Builder::from_env(Env::default().default_filter_or("info"));
+    }
+
+    builder.format(|f, record| match record.level() {
+        log::Level::Error => {
+            eprintln!("{}", record.args());
+            Ok(())
+        }
+        _ => {
+            writeln!(f, "{}", record.args())
+        }
+    });
+
+    builder.target(Target::Stdout).init();
+
+    info!("module version: {}", env!("CARGO_PKG_VERSION"));
+
+    let mut certs_file = std::io::BufReader::new(
+        std::fs::File::open(std::env::var("SSL_CERT_PATH").expect("SSL_CERT_PATH missing"))
+            .expect("read ssl cert"),
+    );
+    let mut key_file = std::io::BufReader::new(
+        std::fs::File::open(std::env::var("SSL_KEY_PATH").expect("SSL_KEY_PATH missing"))
+            .expect("read ssl key"),
+    );
+
+    let tls_certs = rustls_pemfile::certs(&mut certs_file)
+        .collect::<Result<Vec<_>, _>>()
+        .expect("failed to parse cert pem");
+
+    let tls_key = rustls_pemfile::rsa_private_keys(&mut key_file)
+        .next()
+        .expect("no keys found")
+        .expect("invalid key found");
+
+    // set up TLS config options
+    let tls_config = rustls::ServerConfig::builder()
+        .with_no_client_auth()
+        .with_single_cert(tls_certs, rustls::pki_types::PrivateKeyDer::Pkcs1(tls_key))
+        .expect("invalid tls config");
+
+    let Ok(server) = HttpServer::new(move || {
+        App::new()
+            .route("/", web::get().to(index))
+            .route("/token/login", web::post().to(login_token))
+            .route("/token/refresh", web::get().to(refresh_token))
+            .route("/reboot", web::post().to(reboot))
+            .route("/reload-network", web::post().to(reload_network))
+            .service(
+                Files::new(
+                    "/static",
+                    std::fs::canonicalize("static").expect("static folder not found"),
+                )
+                .show_files_listing(),
+            )
+    })
+    .bind_rustls_0_22("0.0.0.0:1977", tls_config) else {
+        error!("cannot bind server");
+        return;
+    };
+
+    let mut centrifugo =
+        Command::new(std::fs::canonicalize("centrifugo").expect("centrifugo not found"))
+            .spawn()
+            .expect("Failed to spawn child process");
+
+    tokio::select! {
+        _ = server.run() => {debug!("1");centrifugo.kill().await.expect("kill failed")},
+        _ = centrifugo.wait() => {}
+    }
```

Comment:
Afaik this pattern makes some issues with SIGKILL, see https://github.com/actix/actix-web/issues/2739

Reply:
I tried to improve.